### PR TITLE
1263 - CVT Call Log Tab

### PR DESCRIPTION
**Pull request by kellykelly3**
Added Call Tag tab to ReleationshipViewer and cleaned\removed the call log code from the message viewer.

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/communications/relationships/RelationshipsNodeUtilities.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (raman-bt) [inline Core/src/org/sleuthkit/autopsy/communications/relationships/RelationshipsNodeUtilities.java]: replace with real file header

File: Core/src/org/sleuthkit/autopsy/communications/relationships/BlackboardArtifactDateComparator.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        BlackboardAttribute attribute1 = null;
+        BlackboardAttribute attribute2 = null;
+        // Inializing to Long.MAX_VALUE so that if a BlackboardArtifact of 
+        // any unexpected type is passed in, it will bubble to the top of 
+        // the list.
+        long dateTime1 = Long.MAX_VALUE;
+        long dateTime2 = Long.MAX_VALUE;
+
+        if (bba1 != null) {
+            BlackboardArtifact.ARTIFACT_TYPE fromID = BlackboardArtifact.ARTIFACT_TYPE.fromID(bba1.getArtifactTypeID());
```
[the remark this answers is not in the document]
Comment (kellykelly3) [reply Core/src/org/sleuthkit/autopsy/communications/relationships/BlackboardArtifactDateComparator.java]: I don't disagree, I was following the convention used in the existing code.  If I change it here there is a bunch of places that I will need to change it.

File: Core/src/org/sleuthkit/autopsy/communications/relationships/CallLogNode.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        String phoneNumber = getAttributeDisplayString(artifact, TSK_PHONE_NUMBER_FROM);
+        if(phoneNumber == null || phoneNumber.isEmpty()) {
+            phoneNumber = getAttributeDisplayString(artifact, TSK_PHONE_NUMBER_TO);
+        }
+        
+        long duration = -1;
+        try{
+            duration = getCallDuration(artifact);
+        } catch(TskCoreException ex) {
+
```
[the remark this answers is not in the document]
Comment (kellykelly3) [reply Core/src/org/sleuthkit/autopsy/communications/relationships/CallLogNode.java]: Because I forgot to add the logger call.

**Comment by kellykelly3** [comment]
I have taken care of all concerns except the fromID.  If I change that code in this class, I should really change it all the other classes, which isn't a big deal, but out of the scope of this jira.  If the use of the name fromID for the value returned from a function called fromID() is of a big enough concern lets open a new jira and I will clean up all of the cvt files that call the fromID function.

**Comment by rcordovano** [comment]
@kellykelly3, please reduce the number of Codacy issues.

**Comment by kellykelly3** [comment]
@rcordovano should be all set.

**Pull request merged**